lib/spack/spack/test/stage.py

```diff
@@ -699,3 +699,37 @@ def test_stage_constructor_with_path(self, tmpdir):
         testpath = str(tmpdir)
         with Stage('file:///does-not-exist', path=testpath) as stage:
             assert stage.path == testpath
+
+    def test_diystage_path_none(self):
+        """Ensure DIYStage for path=None behaves as expected."""
+        with pytest.raises(ValueError):
+            DIYStage(None)
+
+    def test_diystage_path_invalid(self):
+        """Ensure DIYStage for an invalid path behaves as expected."""
+        with pytest.raises(spack.stage.StagePathError):
+            DIYStage('/path/does/not/exist')
+
+    def test_diystage_path_valid(self, tmpdir):
+        """Ensure DIYStage for a valid path behaves as expected."""
+        path = str(tmpdir)
+        stage = DIYStage(path)
+        assert stage.path == path
+        assert stage.source_path == path
+
+        # Order doesn't really matter for DIYStage
+        stage.create()  # Only sets the flag value
+        assert stage.created
+
+        stage.cache_local()  # Only outputs a message
```

Comment:
IMO it's worth putting a message before all these calls stating that they are NOOPs but that they are all called as part of normal stage usage.